delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/auth/AuthTokenProvider.scala

```diff
@@ -0,0 +1,61 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.auth
+
+import cats.effect.Clock
+import ch.epfl.bluebrain.nexus.delta.kernel.cache.KeyValueStore
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.IOUtils
+import ch.epfl.bluebrain.nexus.delta.sdk.identities.model.AuthToken
+import monix.bio.UIO
+
+import java.time.{Duration, Instant}
+
+/**
+  * Provides an auth token for the service account, for use when comunicating with remote storage
+  */
+trait AuthTokenProvider {
+  def apply(): UIO[Option[AuthToken]]
```

Comment:
There is only one set of credentials, so this isn't necessary

Reply:
Don't forget we will have to do the same thing for composite views

Reply:
When we have two then it's an easy refactor, I don't want to design for something that doesn't exist yet

Reply:
For example, one design might be that you have two auth token providers, which wouldn't need that change. I think it's better to change the design when it's needed. It would be a very simple change to make later

Reply:
We don't need two instances, we should avoid that when we can.
The design would look like:
```
trait AuthTokenProvider {
  def apply(credentials: Credentials): UIO[Option[AuthToken]]
}

object AuthTokenProvider {

  def apply(...) = new AuthTokenProvider {
    override def apply(credentials: Credentials): UIO[Option[AuthToken]] =
      credentials match {
        case Anonymous => UIO.none
        case _ => ???
      }
  }

}
```

Reply:
What is the problem with two instances? 

I think a class that does two auth methods is less simple than two classes that are responsible for a single auth method each

Reply:
* You need the authService to instantiate each of them anyway
* When we implement the thing for composite views, it will be easier to integrate in Distage as there will be only one instance of `AuthTokenProvider`